app/scripts/app-init.js

```diff
@@ -127,6 +130,15 @@ chrome.runtime.onMessage.addListener(() => {
   return false;
 });
 
+// Here respond to all clients with ACK_KEEP_ALIVE_MESSAGE
+chrome.runtime.onMessage.addListener((message) => {
+  if (message.name === WORKER_KEEP_ALIVE_MESSAGE) {
+    // eslint-disable-next-line no-undef
+    const channel = new BroadcastChannel('sw-messages');
+    channel.postMessage({ name: ACK_KEEP_ALIVE_MESSAGE });
+  }
+});
```

Comment:
We can add this to listener above [here](https://github.com/MetaMask/metamask-extension/pull/15774/files#diff-cd37b653f9862ccada12e440c6ef67e8b2b2c5b906a320870ed71232d244e660R128)

Reply:
I'm not sure if there's any draw back to having a different listener to handle this separate from the one that handles importAllScripts. I had the logic in one handler, but separating them feels cleaner.

Reply:
Is there any limit on listeners? If the limit is either non-existent or high enough to be essentially non-existent, keep them separate.